src/config/backend.ts

```diff
@@ -424,3 +439,223 @@ export class ExtlinuxConfigBackend extends DeviceConfigBackend {
+		await remountAndWriteAtomic(this.ConfigFilePath, JSON.stringify(config));
+	}
+
+	private async loadConfiguredSsdt(config: ConfigfsConfig): Promise<void> {
+		if (config['ssdt'] !== undefined && _.isArray(config['ssdt'])) {
+			log.info('Loading configured SSDTs');
+			await Bluebird.mapSeries(config['ssdt'], aml => this.loadAML(aml));
+		}
+	}
+
+	public async initialise(
+		opts: BackendOptions,
+	): Promise<ConfigfsConfigBackend> {
+		try {
+			super.initialise(opts);
+
+			// load the acpi_configfs module...
+			await child_process.exec('modprobe acpi_configfs');
+
+			// read the existing config file...
+			const config = await this.readConfigJSON();
+
+			// write the config back out (reformatting it)
+			await this.writeConfigJSON(config);
+
+			// load the configured SSDT AMLs...
+			await this.loadConfiguredSsdt(config);
+			log.success('Initialised ConfigFS');
+		} catch (error) {
+			log.error(error);
+			if (this.options.logger) {
+				this.options.logger.logSystemMessage(
+					'Unable to initialise ConfigFS',
+					{ error },
+					'ConfigFS initialisation error',
+				);
+			}
+		}
+		return this;
+	}
+
+	public matches(deviceType: string): boolean {
+		return ConfigfsConfigBackend.SupportedDeviceTypes.includes(deviceType);
+	}
+
+	public async getBootConfig(): Promise<ConfigOptions> {
+		const options: ConfigOptions = {};
+
+		// read the config file...
+		const config = await this.readConfigJSON();
+
+		// see which SSDTs we have configured...
+		const ssdt = config['ssdt'];
+		if (_.isArray(ssdt) && ssdt.length > 0) {
+			// we have some...
+			options['ssdt'] = ssdt;
+		}
+		return options;
+	}
+
+	public async setBootConfig(opts: ConfigOptions): Promise<void> {
+		// read the config file...
+		const config = await this.readConfigJSON();
+
+		// see if the target state defined some SSDTs...
+		const ssdtKey = `${this.ConfigVarNamePrefix}ssdt`;
+		if (opts[ssdtKey]) {
+			// it did, so update the config with theses...
+			const ssdt = opts[ssdtKey];
+			const ssdts = _.isArray(ssdt) ? ssdt : [ssdt];
+			config['ssdt'] = ssdts;
+		} else {
+			// it did not, so remove any existing SSDTs from the config...
+			delete config['ssdt'];
+		}
+
+		// store the new config to disk...
+		await this.writeConfigJSON(config);
+	}
+
+	public isSupportedConfig(name: string): boolean {
+		switch (this.stripPrefix(name)) {
+			case 'ssdt':
+				return true;
+			default:
+				return false;
+		}
+	}
+
+	public isBootConfigVar(name: string): boolean {
+		switch (this.stripPrefix(name)) {
+			case 'ssdt':
+				return true;
+			default:
+				return false;
+		}
+	}
+
+	public processConfigVarName(name: string): string {
+		return name;
+	}
+
+	public processConfigVarValue(name: string, value: string): string | string[] {
+		switch (this.stripPrefix(name)) {
+			case 'ssdt':
+				// value could be a single value, so just add to an array and return...
+				if (!_.startsWith(value, '"')) {
```

Comment:
This could also be
```suggestion
				if (!value.startsWith('"')) {
```
and just use the native function, assuming the typing is correct